prover/src/domain.rs

```diff
@@ -32,11 +38,37 @@ impl<B: StarkField> StarkDomain<B> {
     /// Returns a new STARK domain initialized with the provided `context`.
     pub fn new<A: Air<BaseField = B>>(air: &A) -> Self {
         let trace_twiddles = fft::get_twiddles(air.trace_length());
+        
+        let domain_gen = B::get_root_of_unity(log2(air.ce_domain_size()));
+        let domain_g = get_power_series(domain_gen, air.ce_domain_size());
+
+        let mut adj_map = BTreeMap::new();
+        let domain_offset = air.domain_offset();
+        let context = air.context();
+        let divisor: ConstraintDivisor<B> = ConstraintDivisor::from_transition(
+            context.trace_len(),
+            context.num_transition_exemptions(),
+        );
+        let div_deg = divisor.degree();
+        let main_constraint_degrees = context.main_transition_constraint_degrees.clone();
+        let aux_constraint_degrees = context.aux_transition_constraint_degrees.clone();
+        let trace_len = context.trace_len();
+        let comp_deg = context.composition_degree();
+        let target_deg = comp_deg + div_deg;
+
+        for degree in main_constraint_degrees.iter().chain(aux_constraint_degrees.iter()){
+            let evaluation_degree = degree.get_evaluation_degree(trace_len);
+            let degree_adjustment = (target_deg - evaluation_degree) as u32;
+            let _ = adj_map.entry(degree_adjustment).or_insert_with(|| {domain_offset.exp(degree_adjustment.into())});
+        }
```

Comment:
I would probably move this logic into a separate helper function.

Also a question: what's the typical number of entries in this map? I think it should be rather small, right?

Reply:
Yes, it would be very small and in the ballpark of the number of `TransitionConstraintGroup`. Thus I am not sure if the BTreeMap is the right DS for this